src/components/sound-browser/SoundSampleGrid.tsx

```diff
@@ -42,27 +48,55 @@ export const SoundSampleGrid = ({
         const sampleKey = `${categoryName}:${index}`;
         const isPlaying = currentlyPlaying === sampleKey;
 
+        // Determine aria-label based on state
+        let ariaLabel: string;
+        if (!canPreview) {
```

Comment:
Please remove if/else statements in this component in favor of ifs with early returns and better error handling.